tutorials/01_introductory/lifecycle.py

```diff
@@ -0,0 +1,285 @@
+# so that they show up more clearly. We can gain access to these labels
+# with the :meth:`axes.Axes.get_xticklabels` method:
+
+fig, ax = plt.subplots()
+ax.barh(group_names, group_data)
+labels = ax.get_xticklabels()
+
+###############################################################################
+# If we'd like to set the property of many items at once, it's useful to use
+# the :func:`pyplot.setp` function. This will take a list (or many lists) of
+# matplotlib objects, and attempt to set some style element of each one.
+
+fig, ax = plt.subplots()
+ax.barh(group_names, group_data)
+labels = ax.get_xticklabels()
+plt.setp(labels, rotation=45, horizontalalignment='right')
+
+###############################################################################
+# It looks like this cut off some of the labels on the bottom. We can
+# tell matplotlib to automatically make room for elements in the figures
+# that we create. To do this we'll set the ``autolayout`` value of our
+# rcParams. For more information on controlling the style, layout, and
+# other features of plots with rcParams, see
+# :ref:`sphx_glr_tutorials_01_introductory_customizing.py`.
+
+plt.rcParams.update({'figure.autolayout': True})
+
+fig, ax = plt.subplots()
+ax.barh(group_names, group_data)
+labels = ax.get_xticklabels()
+plt.setp(labels, rotation=45, horizontalalignment='right')
+
+###############################################################################
+# Next, we'll add labels to the plot. To
+# do this with the OO interface, we can use the :meth:`axes.Axes.set` method
+# to set properties of this axis object.
+
+fig, ax = plt.subplots()
+ax.barh(group_names, group_data)
+labels = ax.get_xticklabels()
+plt.setp(labels, rotation=45, horizontalalignment='right')
+ax.set(xlim=[-10000, 140000], xlabel='Total Revenue', ylabel='Company',
+       title='Company Revenue')
+
+###############################################################################
+# We can also adjust the size of this plot using the :func:`pyplot.subplots`
+# function. We can do this with the ``figsize`` kwarg.
+#
+# .. note::
+#
+#    While indexing in python follows the form (row, column), the figsize
+#    kwarg follows the form (width, height). This follows conventions in
+#    visualization, which unfortunately are different from those of linear
+#    algebra.
+
+fig, ax = plt.subplots(figsize=(8, 4))
+ax.barh(group_names, group_data)
+labels = ax.get_xticklabels()
+plt.setp(labels, rotation=45, horizontalalignment='right')
+ax.set(xlim=[-10000, 140000], xlabel='Total Revenue', ylabel='Company',
+       title='Company Revenue')
+
+###############################################################################
+# For labels, we can specify custom formatting guidelines in the form of
+# functions by using the :class:`ticker.FuncFormatter` class. Below we'll
+# define a function that takes an integer as input, and returns a string
+# as an output.
+
+
+def currency(x, pos):
```

Comment:
Yeah, even I sort of go back and forth on that, 'cause it's awesome for FuncFormatters but possibly yeah too much for intro. Also only know about the decorators 'cause of a discussion on gitter ages ago.